pkg/unpackerr/cnfgfile.go

```diff
@@ -311,12 +312,60 @@ func (u *Unpackerr) validateApp(conf *StarrConfig, app starr.App) error {
 		conf.Protocols = defaultProtocol
 	}
 
+	// Configure TLS and HTTP client
+	tlsConfig, err := u.configureTLS(conf, app)
+	if err != nil {
+		return err
+	}
+
 	conf.Config.Client = &http.Client{
 		Timeout: conf.Timeout.Duration,
 		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: !conf.ValidSSL}, //nolint:gosec
+			TLSClientConfig: tlsConfig,
 		},
 	}
 
 	return nil
 }
+
+// configureTLS creates and configures the TLS config for mTLS support.
+func (u *Unpackerr) configureTLS(conf *StarrConfig, app starr.App) (*tls.Config, error) {
+	// Create TLS config - default behavior unchanged
+	tlsConfig := &tls.Config{InsecureSkipVerify: !conf.ValidSSL} //nolint:gosec
+
+	// Add mTLS if certificates are configured
+	if conf.TLSClientCert != "" && conf.TLSClientKey != "" {
+		certPath := expandHomedir(conf.TLSClientCert)
+		keyPath := expandHomedir(conf.TLSClientKey)
+
+		cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+		if err != nil {
+			return nil, fmt.Errorf("%s (%s) failed loading TLS client cert from %s and %s: %w",
+				app, conf.URL, certPath, keyPath, err)
+		}
+
+		tlsConfig.Certificates = []tls.Certificate{cert}
+
+		u.Debugf("%s (%s): Loaded mTLS client certificate", app, conf.URL)
+	}
+
+	// Add custom CA if configured
+	if conf.TLSCACert != "" {
+		caCert, err := os.ReadFile(expandHomedir(conf.TLSCACert))
+		if err != nil {
+			return nil, fmt.Errorf("%s (%s) failed reading CA cert from %s: %w",
+				app, conf.URL, expandHomedir(conf.TLSCACert), err)
+		}
+
+		caCertPool := x509.NewCertPool()
```

Comment:
```suggestion
		tlsConfig.RootCA = x509.NewCertPool()
```

Can you just set this directly here? (I'm not sure if .RootCAs is an interface or not.)